juicy-select.html

```diff
@@ -27,45 +27,41 @@
                 valueProperty: { type: String, observer: "optionsChanged" },
                 captionText: { type: String, observer: "optionsChanged" }
             },
+            created: function () {
+                this.addEventListener("change", this.selectChange.bind(this));
+            },
             attached: function () {
                 this.$select = this.querySelector("select");
-
                 if (!this.$select) {
                     this.$select = document.createElement("select");
                     this.appendChild(this.$select);
-
                     if (this.multiple) {
                         this.$select.setAttribute("multiple", "multiple");
                     }
                 }
-
-                this.$select.addEventListener("change", this.selectChange.bind(this));
                 this.set("multiple", this.$select.hasAttribute("multiple"));
-
-                setTimeout(function () {
-                    this.generateOptions();
-                    this.isAttached = true;
-
-                    if (this.value != this.$select.value && !this.multiple) {
-                        this.set("value", this.$select.value);
-                    }
-                }.bind(this), 10);
             },
             optionsChanged: function () {
-                if (this.isAttached) {
-                    this.generateOptions();
+                if (!this.$select) {
+                    return;
+                }
+
+                this.generateOptions();
+
+                if (this.value != this.$select.value && !this.multiple) {
+                    this.set("value", this.$select.value);
                 }
             },
             generateOptions: function () {
+                if (!this.options || !this.options.length) {
```

Comment:
In this case the empty option or caption text will not be shown when there are no options available, but it should be.

Reply:
Fixed. I thought this is not intended use case, but I agree I should not make such breaking changes.